Remotion/Validation/UnitTests/RuleBuilders/AddingObjectValidationRuleBuilderTest.cs

```diff
@@ -14,61 +14,153 @@
+      _addingObjectValidationRuleCollectorMock
+          .Setup(stub => stub.ValidatedType)
+          .Returns(TypeAdapter.Create(typeof(Customer)));
+
+      _addingObjectValidationBuilder = new AddingObjectValidationRuleBuilder<Customer>(
+          _addingObjectValidationRuleCollectorMock.Object,
+          _objectMetaValidationRuleCollectorMock.Object);
+
+      _objectValidatorStub = new Mock<IObjectValidator>();
+    }
 
     [Test]
-    [Ignore("TODO RM-5906")]
     public void Initialization ()
     {
+      Assert.That(_addingObjectValidationBuilder.AddingObjectValidationRuleCollector, Is.SameAs(_addingObjectValidationRuleCollectorMock.Object));
+      Assert.That(_addingObjectValidationBuilder.ObjectMetaValidationRuleCollector, Is.SameAs(_objectMetaValidationRuleCollectorMock.Object));
     }
 
     [Test]
-    [Ignore("TODO RM-5906")]
     public void SetValidator ()
     {
+      Func<ObjectValidationRuleInitializationParameters, IObjectValidator> validatorFactory = _ => _objectValidatorStub.Object;
+      _addingObjectValidationRuleCollectorMock
+          .Setup(mock => mock.RegisterValidator(validatorFactory))
+          .Verifiable();
+
+      ((IAddingObjectValidationRuleBuilder<Customer>)_addingObjectValidationBuilder).SetValidator(validatorFactory);
+
+      _addingObjectValidationRuleCollectorMock.Verify();
     }
 
     [Test]
-    [Ignore("TODO RM-5906")]
     public void CanBeRemoved ()
     {
+      _addingObjectValidationRuleCollectorMock
+          .Setup(mock => mock.SetRemovable())
+          .Verifiable();
+
+      _addingObjectValidationBuilder.CanBeRemoved();
+
+      _addingObjectValidationRuleCollectorMock.Verify();
     }
 
     [Test]
-    [Ignore("TODO RM-5906")]
     public void AddMetaValidationRule ()
     {
+      var metaValidationRuleStub = new Mock<IObjectMetaValidationRule>();
+
+      _objectMetaValidationRuleCollectorMock
+          .Setup(mock => mock.RegisterMetaValidationRule(metaValidationRuleStub.Object))
+          .Verifiable();
+
+      _addingObjectValidationBuilder.AddMetaValidationRule(metaValidationRuleStub.Object);
+
+      _objectMetaValidationRuleCollectorMock.Verify();
     }
 
     [Test]
-    [Ignore("TODO RM-5906")]
     public void AddMetaValidationRule_FuncOverload ()
     {
+      var fakeValidationResult = MetaValidationRuleValidationResult.CreateValidResult();
+
+      IObjectMetaValidationRule rule = null;
+      _objectMetaValidationRuleCollectorMock
+          .Setup(mock => mock.RegisterMetaValidationRule(It.IsAny<IObjectMetaValidationRule>()))
+          .Callback((IObjectMetaValidationRule r) => rule = r);
+
+      _addingObjectValidationBuilder.AddMetaValidationRule(v => fakeValidationResult);
+
+      Assert.That(rule.GetType(), Is.EqualTo(typeof(DelegateObjectMetaValidationRule<IObjectValidator>)));
+      Assert.That(rule.Validate(new IObjectValidator[0]).First(), Is.EqualTo(fakeValidationResult));
```

Comment:
```suggestion
      Assert.That(rule, Is.TypeOf<DelegateObjectMetaValidationRule<IObjectValidator>>));
      Assert.That(rule.Validate(new IObjectValidator[0]), Is.EquivalentTo(new[]{ fakeValidationResult }));
```
In both cases, we use native NUnit operations to do the assertions, getting a better error message in case of red tests. in the second case, we want to check a result collection. this can be done via EqualTo or EquivalentTo, the difference that EqualTo also asserts position, not just value. Since the order of the result items isn't relevant here, EquivalentTo is best. By using First, we only test that there's a first item, not that there could also be others. In total, the updated version is more robust and has better diagnostics.

Please check the rest of the PR for similiar cases.

Reply:
There seems to be an error. I think you missed `typeof()` in the first Assert:
`Assert.That(rule, Is.TypeOf(typeof(DelegateObjectMetaValidationRule<IObjectValidator>)));`

Reply:
@KuJo7, actually, I was using Generics syntax and missed the method call parens at the end.